mtags/src/main/scala-3/scala/meta/internal/pc/HoverProvider.scala

```diff
@@ -29,6 +29,36 @@ object HoverProvider:
       driver: InteractiveDriver,
       search: SymbolSearch,
   )(implicit reportContext: ReportContext): ju.Optional[HoverSignature] =
+    def dealias(tpe: Type)(using Context): Type = {
```

Comment:
We have a method called `metalsDealias` (it should be named deepDealias) which does similar thing to yours.
The fix should be done directly in that method.
This is a bit more controlled of what we're doing.

I recommend implementing this like that:
```scala
  extension (tpe: Type)
    def deepDealias(using Context): Type =
      tpe.dealias match
        case app @ AppliedType(tycon, params) =>
          // we dealias applied type params by hand, because `dealias` doesn't do it
          AppliedType(tycon, params.map(_.metalsDealias))
        case aliasingBounds: AliasingBounds => 
          aliasingBounds.derivedAlias(aliasingBounds.alias.dealias)
        case TypeBounds(lo, hi) => 
          TypeBounds(lo.dealias, hi.dealias)
        case RefinedType(parent, name, refinedInfo) => 
          RefinedType(parent.dealias, name, refinedInfo.deepDealias)
        case dealised => dealised
```

Reply:
Hm.. I am getting a strange error in one of the java import tests. I haven't seen this before. Is this something known? This is 3.3.3. Thanks!

exception caught when loading class BoundMethodHandle$Specializer$Factory: java.lang.AssertionError: assertion failed: failure to resolve inner class:
externalName = java.lang.invoke.ClassSpecializer$Factory,
outerName = java.lang.invoke.ClassSpecializer,
innerName = Factory
owner.fullName = java.lang.invoke.ClassSpecializer
while parsing /modules/java.base/java/lang/invoke/BoundMethodHandle$Specializer$Factory.class
Mar 27, 2024 11:13:19 PM scala.meta.internal.pc.CompilerAccess handleError
SEVERE: assertion failed: failure to resolve inner class:
externalName = java.lang.invoke.ClassSpecializer$Factory,
outerName = java.lang.invoke.ClassSpecializer,
innerName = Factory
owner.fullName = java.lang.invoke.ClassSpecializer
while parsing /modules/java.base/java/lang/invoke/BoundMethodHandle$Specializer$Factory.class
java.lang.AssertionError: assertion failed: failure to resolve inner class:
externalName = java.lang.invoke.ClassSpecializer$Factory,
outerName = java.lang.invoke.ClassSpecializer,
innerName = Factory
owner.fullName = java.lang.invoke.ClassSpecializer
while parsing /modules/java.base/java/lang/invoke/BoundMethodHandle$Specializer$Factory.class
	at scala.runtime.Scala3RunTime$.assertFailed(Scala3RunTime.scala:8)
	at dotty.tools.dotc.core.classfile.ClassfileParser$innerClasses$.classSymbol(ClassfileParser.scala:1124)

Reply:
Are you getting this on this branch ? Maybe try cleaning the project `sbt clean` and try again ?